Description: Support Mint OS detection
 mate-system-monitor (1.10.0-2) rafaela; urgency=medium
 .
   * Updated packaging
   * Show proper distro info
Author: Clement Lefebvre <root@linuxmint.com>


--- mate-system-monitor-1.10.0.orig/src/sysinfo.cpp
+++ mate-system-monitor-1.10.0/src/sysinfo.cpp
@@ -544,6 +544,35 @@ namespace {
         }
     };
 
+    class MintSysInfo
+        : public SysInfo
+    {
+    public:
+        MintSysInfo()
+        {
+            this->load_os_release();
+        }
+
+    private:
+        void load_os_release()
+        {
+            this->distro_name = "Linux Mint";
+            std::ifstream input("/etc/linuxmint/info");
+            if (input) {
+                while (!input.eof()) {
+                    string s;
+                    int len;
+                    std::getline(input, s);
+                    if (s.find("DESCRIPTION=") == 0) {
+                        len = strlen("DESCRIPTION=");
+                        // also strip the surrounding quotes
+                       this->distro_release = s.substr(len + 1, s.size() - len - 2);
+                   }
+                }
+            }
+        }
+    };
+
 
     class OpenBSDSysInfo
         : public SysInfo
@@ -569,6 +598,9 @@ namespace {
 
     SysInfo* get_sysinfo()
     {
+        if (g_file_test ("/etc/linuxmint/info", G_FILE_TEST_EXISTS)) {
+            return new MintSysInfo;
+        }
         if (g_file_test ("/etc/os-release", G_FILE_TEST_EXISTS)) {
             return new GenericSysInfo;
         }
